tensorzero-internal/tests/e2e/providers/common.rs

```diff
@@ -446,25 +450,54 @@ pub async fn test_extra_body_with_provider(provider: E2ETestProvider) {
                     "content": "What is the name of the capital city of Japan?"
                 }
             ]},
-        "stream": false,
+        "stream": stream,
         "tags": {"foo": "bar"},
     });
 
-    let response = Client::new()
-        .post(get_gateway_endpoint("/inference"))
-        .json(&payload)
-        .send()
-        .await
-        .unwrap();
+    let inference_id = if stream {
+        let mut event_source = Client::new()
+            .post(get_gateway_endpoint("/inference"))
+            .json(&payload)
+            .eventsource()
+            .unwrap();
 
-    // Check that the API response is ok
-    assert_eq!(response.status(), StatusCode::OK);
-    let response_json = response.json::<Value>().await.unwrap();
+        let mut chunks = vec![];
+        let mut found_done_chunk = false;
+        while let Some(event) = event_source.next().await {
+            let event = event.unwrap();
+            match event {
+                Event::Open => continue,
+                Event::Message(message) => {
+                    if message.data == "[DONE]" {
+                        found_done_chunk = true;
+                        break;
+                    }
+                    chunks.push(message.data);
+                }
+            }
+        }
+        assert!(found_done_chunk);
 
-    println!("API response: {response_json:#?}");
+        let response_json = serde_json::from_str::<Value>(&chunks[0]).unwrap();
+        let inference_id = response_json.get("inference_id").unwrap().as_str().unwrap();
+        Uuid::parse_str(inference_id).unwrap()
+    } else {
+        let response = Client::new()
+            .post(get_gateway_endpoint("/inference"))
+            .json(&payload)
+            .send()
+            .await
+            .unwrap();
 
-    let inference_id = response_json.get("inference_id").unwrap().as_str().unwrap();
-    let inference_id = Uuid::parse_str(inference_id).unwrap();
+        // Check that the API response is ok
+        assert_eq!(response.status(), StatusCode::OK);
+        let response_json = response.json::<Value>().await.unwrap();
+
+        println!("API response: {response_json:#?}");
+
+        let inference_id = response_json.get("inference_id").unwrap().as_str().unwrap();
+        Uuid::parse_str(inference_id).unwrap()
+    };
```

Comment:
hey I think you forgot (previously) to test the actual ChatInference table, let's add that

Reply:
or otherwise, add a comment explaining that it is not interesting for this test and we can skip